lib/resque_bus/subscriber.rb

```diff
@@ -0,0 +1,28 @@
+module QueueBus
+  module Subscriber
+
+    # Store the QueueBus implementation of perform so we can wrap it
+    @base_perform = ClassMethods.instance_method(:perform)
+
+    module ClassMethods
+      # Wrap the QueueBus implementation of perform so the calls to Resque
+      # invoke the local Resque instance, not that used by QueueBus
+      def perform(attributes)
+        with_local_redis do
```

Comment:
This looks the right idea, but you're doing it only in subscriber.
I think it also needs to happen in `Rider` and maybe more.

One thought is to do it in Worker, which is the runner for all of them.
I believe you do it just in the `Adapter` with a pattern similiar to these `RetryHandlers` there.

Your mixin would either:
1) add two Resque callbacks (before and ensured after)
2) override the class method similar to what you've done here.